crates/uplc/src/program_builder/tests.rs

```diff
@@ -1,33 +1,65 @@
 use super::*;
 use crate::parser;
 use crate::program_builder::constant::WithConstant;
-use proptest::prelude::*;
+use quickcheck::{Arbitrary, Gen};
 
-prop_compose! {
-    fn arb_version()(
-        maj: isize,
-        min: isize,
-        patch: isize,
-    ) -> (usize, usize, usize){
-        let maj = maj.unsigned_abs();
-        let min = min.unsigned_abs();
-        let patch = patch.unsigned_abs();
+// #[derive(Clone, Debug)]
+// struct Version {
+//     maj: isize,
+//     min: isize,
+//     patch: isize,
+// }
+//
+// impl Version {
+//     fn inner(&self) -> (isize, isize, isize) {
+//         (self.maj, self.min, self.patch)
+//     }
+// }
+//
+// impl Arbitrary for Version {
+//     fn arbitrary(g: &mut Gen) -> Self {
+//         let maj = isize::arbitrary(g).unsigned_abs() as isize;
+//         let min = isize::arbitrary(g).unsigned_abs() as isize;
+//         let patch = isize::arbitrary(g).unsigned_abs() as isize;
+//         Version { maj, min, patch }
+//     }
+// }
 
-        (maj, min, patch)
+#[derive(Clone, Debug)]
+struct Version {
+    maj: usize,
+    min: usize,
+    patch: usize,
+}
+
+impl Version {
+    fn inner(&self) -> (usize, usize, usize) {
```

Comment:
If inner returns `usize` this fails. If it returns `isize` like above it passes, I just have to add some `as`s to the test.